### ISSUE TITLE
Incorrect gradient for linalg.cholesky

### ISSUE THREAD

**Issue by sbodenstein** [jax-ml/jax#40421]
### Description

Consider this on CPU:
```
import jax
import jax.numpy as jnp
import jax.scipy as jsp

x = jnp.array([[4.0, 1.0], [1.0, 3.0]])
loss_fn = lambda x: jnp.sum(jsp.linalg.cholesky(x, lower=False))
jax.grad(loss_fn)(x)
```
gives 
```
Array([[ 0.20634446,  0.42462218],
       [-0.07537784,  0.30151135]], dtype=float32)
```
With `lower=False`, the lower-triangular parts are ignored, which is correctly done in the forward pass. But a ghost-gradient is introduced in the backwards (the gradient element (1, 0) should be zero as the value is ignored but is `-0.07537784`).

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.11.2
jaxlib: 0.11.2
numpy:  2.5.2
```
